index.js

```diff
@@ -337,19 +338,55 @@ module.exports = class BlindPeering {
   _addBaseCores(ref, base, all) {
     if (this.passive) return Promise.all([ref.peer.connect()])
 
-    const promises = []
-    promises.push(this._mirrorBaseWriter(ref, base, base.local, true))
+    const coresToAdd = new Map()
+    if (!ref.cores.has(base.local.id)) coresToAdd.set(base.local.id, base.local)
 
+    // Add up to 'autobaseWritersPerRequest' active writers (random)
+    const candidates = []
     for (const writer of base.activeWriters) {
-      const always = isStaticCore(writer.core) || all
-      promises.push(this._mirrorBaseWriter(ref, base, writer.core, always))
+      if (!coresToAdd.has(writer.core.id)) candidates.push(writer.core)
     }
-
-    for (const view of base.views()) {
-      promises.push(ref.peer.addCore(view.key, { announce: false, referrer: null, priority: 1 }))
+    const sliceStart = Math.floor(Math.random() * candidates.length)
+    for (let i = sliceStart; i < candidates.length; i++) {
+      if (coresToAdd.size >= this.autobaseWritersPerRequest) break
+      const core = candidates[i]
+      coresToAdd.set(core.id, core)
+    }
+    for (let i = 0; i < sliceStart; i++) {
+      if (coresToAdd.size >= this.autobaseWritersPerRequest) break
+      const core = candidates[i]
+      coresToAdd.set(core.id, core)
     }
 
-    return Promise.all(promises)
+    ref.refs++
+    try {
+      for (const core of coresToAdd.values()) {
+        // TODO: if the blind-peer server does not set up replication, because
```

Comment:
Clarification needed on what we should do for these cases (ATM we can't even detect when the blind peer doesn't set up replication for this core, but if necessary we could return that from the RPC call)